src/NewsService.Validation/EditNewsRequestValidator.cs

```diff
@@ -0,0 +1,194 @@
+        x => x == OperationType.Replace,
+        new()
+        {
+          { x => x.value.ToString().Length < 30, "Pseudonym is too long." },
+        });
+
+      #endregion
+
+      #region AuthorId, DepartmentId
+
+      AddFailureForPropertyIf(
+          nameof(EditNewsRequest.DepartmentId),
+          x => x == OperationType.Replace,
+          new()
+          {
+            { x => Guid.TryParse(x.value.ToString(), out Guid result), "Department id has incorrect format" },
+            { x => CheckDepartmentExistence(new List<Guid> { Guid.Parse(x.value.ToString()) }), "This department doesn't exist." }
+          });
+
+      AddFailureForPropertyIf(
+        nameof(EditNewsRequest.AuthorId),
+        x => x == OperationType.Replace,
+        new()
+        {
+          { x => Guid.TryParse(x.value.ToString(), out Guid result), "User id has incorrect format" },
+          { x => CheckUserExistence(new List<Guid> { Guid.Parse(x.value.ToString()) }), "This user doesn't exist." }
+        });
+
+      #endregion
+
+      #region IsActive
+
+      AddFailureForPropertyIf(
+        nameof(EditNewsRequest.IsActive),
+        x => x == OperationType.Replace,
+        new()
+        {
+          { x => bool.TryParse(x.value?.ToString(), out _), "Incorrect is active format." },
+        });
+
+      #endregion
+    }
+    public EditNewsRequestValidator(
+      IRequestClient<ICheckUsersExistence> rcCheckUsersExistence,
+      IRequestClient<ICheckDepartmentsExistence> rcCheckDepartmentsExistence,
+      ILogger<CreateNewsRequestValidator> logger)
+    {
+      _rcCheckUsersExistence = rcCheckUsersExistence;
+      _rcCheckDepartmentsExistence = rcCheckDepartmentsExistence;
+      _logger = logger;
+
+      RuleForEach(x => x.Operations)
+        .Custom(HandleInternalPropertyValidation);
+    }
+
+    private bool CheckUserExistence(List<Guid> authorsIds)
+    {
+      if (!authorsIds.Any() || authorsIds == default)
+      {
+        return false;
+      }
+
+      try
+      {
+        Response<IOperationResult<ICheckUsersExistence>> response =
+          _rcCheckUsersExistence.GetResponse<IOperationResult<ICheckUsersExistence>>(
+          ICheckUsersExistence.CreateObj(authorsIds)).Result;
+
+        if (response.Message.IsSuccess)
+        {
+          return authorsIds.Count == response.Message.Body.UserIds.Count;
+        }
+
+        _logger.LogWarning("Can not find author Ids: {authorsIds}: " +
+          $"{Environment.NewLine}{string.Join('\n', response.Message.Errors)}");
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, "Cannot check existing authors withs this ids {authorsIds}");
+      }
+
+      return false;
+    }
+
+    private bool CheckDepartmentExistence(List<Guid> departmentsIds)
+    {
+      if (!departmentsIds.Any() ||  departmentsIds == default)
+      {
+        return false;
+      }
+
+      try
+      {
+        Response<IOperationResult<ICheckDepartmentsExistence>> response =
```

Comment:
same

Reply:
@MariyaSin said to leave it as it is for now